### PR TITLE
Add via support for 6key

### PR DESCRIPTION
**Pull request by Bratzworth**
Adds via support for the 6key macropad

## Description

Adds a new keymap to the 6key macropad, and moves the dip switch configuration from the default keymap to `6key.c`. As a part of adding VIA support the vendor id was changed to be unique.

Also appropriately updates all references from LAYOUT to LAYOUT_all.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/handwired/6key/config.h [hunk 1]
```diff
@@ -1,23 +1,31 @@
...
 #include "config_common.h"
 
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0xD143
+#define PRODUCT_ID      0x0007
+#define DEVICE_VER      0x0002
+
+#define MANUFACTURER    bratzworth
+#define PRODUCT         6key
+
```
Comment (fauxpark) [inline keyboards/handwired/6key/config.h:28]:
```suggestion
```

File: keyboards/handwired/6key/info.json [hunk 2]
```diff
@@ -1,16 +1,19 @@
...
-    "manufacturer": "Bratzworth",
+    "manufacturer": "bratzworth",
     "url": "https://github.com/Bratzworth/6key",
     "maintainer": "bratzworth",
     "usb": {
-      "vid": "0xBEED",
+      "vid": "0xD143",
       "pid": "0x0007",
-      "device_version": "0.0.1"
+      "device_version": "0.0.2"
```
Comment (fauxpark) [inline keyboards/handwired/6key/info.json:9]:
```suggestion
      "device_version": "0.0.1"
```

File: keyboards/handwired/6key/keymaps/via/readme.md [hunk 3]
```diff
@@ -0,0 +1 @@
+# The via keymap for the 6key
```
Comment (fauxpark) [inline keyboards/handwired/6key/keymaps/via/readme.md:1]: This file can be deleted.

File: keyboards/handwired/6key/keymaps/default/readme.md [hunk 4]
```diff
@@ -0,0 +1 @@
+# The default keymap for the 6key
```
Comment (fauxpark) [inline keyboards/handwired/6key/keymaps/default/readme.md:1]: This file can be deleted.

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Comment by Bratzworth** [comment]
is there any way to request a re-review of this?

File: keyboards/handwired/6key/rules.mk [hunk 5]
```diff
@@ -7,8 +7,8 @@ BOOTLOADER = caterina
 # Build Options
 #   change yes to no to disable
 #
-BOOTMAGIC_ENABLE = no       # Enable Bootmagic Lite
-MOUSEKEY_ENABLE = yes       # Mouse keys
+BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = no        # Mouse keys
```
Comment (zvecr) [inline keyboards/handwired/6key/rules.mk:11]:
```suggestion
MOUSEKEY_ENABLE = yes       # Mouse keys
```

File: keyboards/handwired/6key/keymaps/via/keymap.c [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+        KC_F16, KC_F17, KC_F18
+    ),
+    [2] = LAYOUT_all(
+        KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS
+        ),
+    [3] = LAYOUT_all(
+        KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS
+        ),
```
Comment (zvecr) [inline keyboards/handwired/6key/keymaps/via/keymap.c:39]:
```suggestion
const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
    [0] = LAYOUT(
        LCTL(KC_Z), LCTL(KC_Y), KC_A,
        KC_C,       KC_D,       KC_S
    ),
    [1] = LAYOUT(
        KC_F13, KC_F14, KC_F15,
        KC_F16, KC_F17, KC_F18
    ),
    [2] = LAYOUT(
        KC_TRNS, KC_TRNS, KC_TRNS,
        KC_TRNS, KC_TRNS, KC_TRNS
        ),
    [3] = LAYOUT(
        KC_TRNS, KC_TRNS, KC_TRNS,
        KC_TRNS, KC_TRNS, KC_TRNS
        ),
```

File: keyboards/handwired/6key/keymaps/default/keymap.c [hunk 7]
```diff
@@ -15,33 +15,18 @@
...
+enum layers{
+  _MAIN,
+  _FN
+};
 
 const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
-  [_MAIN] = LAYOUT(
-    KC_UNDO,  KC_REDO,  KC_A,
-    KC_C,     KC_D,     KC_S
+  [_MAIN] = LAYOUT_all(
```
Comment (zvecr) [inline keyboards/handwired/6key/keymaps/default/keymap.c:24]:
```suggestion
  [_MAIN] = LAYOUT(
```

File: keyboards/handwired/6key/keymaps/default/keymap.c [hunk 8]
```diff
@@ -15,33 +15,18 @@
...
+    KC_Z, KC_Y, KC_A,
+    KC_C, KC_D, KC_S
   ),
-
-  [_FN] = LAYOUT(
+  [_FN] = LAYOUT_all(
```
Comment (zvecr) [inline keyboards/handwired/6key/keymaps/default/keymap.c:28]:
```suggestion
  [_FN] = LAYOUT(
```

File: keyboards/handwired/6key/info.json [hunk 9]
```diff
@@ -1,16 +1,19 @@
...
     "usb": {
-      "vid": "0xBEED",
+      "vid": "0xD143",
       "pid": "0x0007",
       "device_version": "0.0.1"
     },
     "layouts": {
-      "LAYOUT": {
-        "layout": [{"label":"k00", "x":0, "y":0}, {"label":"k01", "x":1, "y":0}, {"label":"k02", "x":2, "y":0}, {"label":"k10", "x":0, "y":1}, {"label":"k11", "x":1, "y":1}, {"label":"k12", "x":2, "y":1}]
+      "LAYOUT_all": {
```
Comment (zvecr) [inline keyboards/handwired/6key/info.json:12]:
```suggestion
      "LAYOUT": {
```

File: keyboards/handwired/6key/6key.h [hunk 10]
```diff
@@ -1,27 +1,27 @@
...
+  *
+  * You should have received a copy of the GNU General Public License
+  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
   */
 #pragma once
 
 #include "quantum.h"
 
-#define LAYOUT( \
+#define LAYOUT_all( \
```
Comment (zvecr) [inline keyboards/handwired/6key/6key.h:20]:
```suggestion
#define LAYOUT( \
```

**Review by zvecr — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Review by zvecr — APPROVED** [review]

**Pull request merged**